platform/msm_shared/include/ab_partition_parser.h

```diff
@@ -79,7 +79,11 @@ int partition_fill_partition_meta(char has_slot_pname[][MAX_GET_VAR_NAME_SIZE],
 				  char has_slot_reply[][MAX_RSP_SIZE],
 				  int array_size);	/* Fill partition slot info meta*/
 
-static inline bool partition_multislot_is_supported(void)
-{
-	return target_is_emmc_boot() && _partition_multislot_is_supported();
+static inline bool partition_multislot_is_supported(void){
+#if SLOTS_SUPPORTED
+	return true;
+#else
+	return false;
+#endif
 }
+				  
```

Comment:
Do I understand correctly that `SLOTS_SUPPORTED` "forces" slots on instead of relying on caf lk's code that detects the presence of blah_a and blah_b? In this case, if this is just an lk2nd extension, please rename it to something like `LK2ND_FORCE_AB_SLOTS` and keep existing behavior if it's not enabled. Do I understand correctly that you need it due to dynamic partitions?

(...) thinking of this more I'm not sure how to integrate this into the menu nicely....

Reply:
I need to look into this again either the detection code was just plainly broken or didn't work for me it was some weeks ago I fixed the code.

Iirc it was badly ported over from edk2's aboot impl (probably taken from shift phone xbl)